components/inputs/input-number.js

```diff
@@ -81,32 +70,69 @@ class InputNumber extends LitElement {
 	}
 
 	render() {
-		const ariaRequired = this.required ? 'true' : undefined;
-
 		return html`
-			<label
-				class="${this.label && !this.labelHidden ? 'd2l-input-label' : 'd2l-offscreen'}"
-				for="${this._inputId}">${this.label}</label>
-			<input
-				aria-required="${ifDefined(ariaRequired)}"
+			<d2l-input-text
 				autocomplete="${ifDefined(this.autocomplete)}"
 				?autofocus="${this.autofocus}"
 				@change="${this._handleChange}"
-				class="d2l-input"
+				@d2l-form-element-should-validate="${this._handleNestedFormElementValidation}"
 				?disabled="${this.disabled}"
+				.forceInvalid="${this.invalid}"
 				id="${this._inputId}"
+				label="${this.label}"
+				?label-hidden="${this.labelHidden}"
 				name="${ifDefined(this.name)}"
 				placeholder="${ifDefined(this.placeholder)}"
-				type="text"
-				.value="${this._formattedValue}">
+				?required="${this.required}"
+				.value="${this._formattedValue}"
+			></d2l-input-text>
+			${ this.validationError ? html`<d2l-tooltip for=${this._inputId} state="error" align="start">${this.validationError}</d2l-tooltip>` : null }
 		`;
 	}
 
+	focus() {
+		this.shadowRoot.querySelector('d2l-input-text').focus();
+	}
+
+	async validate(showErrors) {
+		const errors = await super.validate(showErrors);
+		if (errors.length !== 0) {
+			return errors;
+		}
+		return await this.shadowRoot.querySelector('d2l-input-text').validate(showErrors);
+	}
+
+	get validationMessage() {
+		if (this.validity.rangeOverflow || this.validity.rangeUnderflow) {
+			const label = this.label && !this.labelHidden ? this.label : 'Number';
```

Comment:
What's the thinking here behind using "Number" if they've chosen to hide the label? I think we always just want to use the label they've given OR just embed "Number" inside the lang term like Allan had for the existing range under/overflow terms -- i.e. `"Number must be lower than {max}"`.

Reply:
Hmm in that case I think it'd make more sense to just use the given label because it'd be clearer to the user.

On the other hand, I wanted to use "Number" if there was no label because the user may not know what the validation error was referring to if a field had a hidden label.

Reply:
When @geurts and I discussed it he mentioned only wanting `Number` because we want to avoid using the field label unless its absolutely necessary. It might be worth double checking just in case something has changed since then.

Reply:
Yeah I think I agree with Jeff here... "Number must be greater than 4" when shown as a tooltip next to any input seems pretty clear to me. When we transplant arbitrary labels into lang terms we run the risk of running into all kinds of weird issues in other languages.